pkg/config/configuration.go

```diff
@@ -206,3 +276,272 @@ func containsKey(s []string, key string) bool {
+			return "", err
+		}
+
+		tlsInfo := peer.AuthInfo.(credentials.TLSInfo)
+
+		oid := asn1.ObjectIdentifier{2, 5, 29, 17}
+
+		for _, crt := range tlsInfo.State.PeerCertificates {
+			for _, ext := range crt.Extensions {
+				if ext.Id.Equal(oid) {
+					var sequence asn1.RawValue
+					if rest, err := asn1.Unmarshal(ext.Value, &sequence); err != nil {
+						log.Error(err)
+						continue
+					} else if len(rest) != 0 {
+						log.Error("the SAN extension is incorrectly encoded")
+						continue
+					}
+
+					if !sequence.IsCompound || sequence.Tag != asn1.TagSequence || sequence.Class != asn1.ClassUniversal {
+						log.Error("the SAN extension is incorrectly encoded")
+						continue
+					}
+
+					for bytes := sequence.Bytes; len(bytes) > 0; {
+						var rawValue asn1.RawValue
+						var err error
+
+						bytes, err = asn1.Unmarshal(bytes, &rawValue)
+						if err != nil {
+							return "", err
+						}
+
+						spiffeID = string(rawValue.Bytes)
+						if strings.HasPrefix(spiffeID, SpiffeIDPrefix) {
+							log.Errorf("Found spiffe id: %s\n", spiffeID)
+							return spiffeID, nil
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return "", nil
+}
+
+// IsOperationAllowedByAccessControlPolicy determines if access control policies allow the operation on the target app
+func IsOperationAllowedByAccessControlPolicy(spiffeID *SpiffeID, srcAppID string, inputOperation string, httpVerb common.HTTPExtension_Verb, accessControlList *AccessControlList) (bool, string) {
+	if accessControlList == nil {
+		// No access control list is provided. Do nothing
+		return isActionAllowed(AllowAccess), ""
+	}
+
+	action := accessControlList.DefaultAction
+	actionPolicy := ActionPolicyGlobal
+
+	if srcAppID == "" {
+		// Did not receive the src app id correctly
+		return isActionAllowed(action), actionPolicy
+	}
+
+	// Look up the src app id in the in-memory table
+	appPolicy, found := accessControlList.PolicySpec[srcAppID]
+
+	if !found {
+		// no policies found for this src app id. Apply global default action
+		return isActionAllowed(action), actionPolicy
+	}
+
+	if spiffeID == nil {
+		// Could not retrieve spiffe id or it is invalid. Apply global default action
+		return isActionAllowed(action), actionPolicy
+	}
+
+	// Match trust domain
+	if appPolicy.TrustDomain != spiffeID.TrustDomain {
+		return isActionAllowed(action), actionPolicy
+	}
+
+	// Match namespace
+	if appPolicy.Namespace != spiffeID.Namespace {
+		return isActionAllowed(action), actionPolicy
+	}
+
+	if appPolicy.DefaultAction != "" {
+		// Since the app has specified a default action, this point onwards,
+		// default action is the default action specified in the spec for the app
+		action = appPolicy.DefaultAction
+		actionPolicy = ActionPolicyApp
+	}
+
+	// the in-memory table has operations stored in the format "/operation name".
+	// Prepend a "/" in case missing so that the match works
+	if !strings.HasPrefix(inputOperation, "/") {
+		inputOperation = "/" + inputOperation
```

Comment:
Again, that's fine so long as we can work with method names that do not contain "/".